test/testscript.go

```diff
@@ -153,41 +163,72 @@ func (e *TestscriptTest) Run(scriptFile string) {
 	}
 	e.params.Files = []string{scriptFile}
 	e.params.Cmds = map[string]func(*testscript.TestScript, bool, []string){
-		"setEnvValueFromFile": cmdSetEnvValueFromFile,
+		"saveFile":            saveFileCmd("saveFile", e.saveDir),
+		"setEnvValueFromFile": setEnvValueFromFileCmd("setEnvValueFromFile"),
 	}
 	testscript.Run(e.t, e.params)
 }
 
-// setEnvValueFromFile envKey filePath
-// setEnvValueFromFile reads the contents of a file (filePath) and sets an
-// environment variable (envKey) to the rendered contents of the file
-func cmdSetEnvValueFromFile(
-	ts *testscript.TestScript,
-	neg bool,
-	args []string,
-) {
-	if neg {
-		ts.Fatalf("unsupported: ! setEnvValueFromFile")
-	}
-	var srcRelPath string
-	if len(args) != 2 {
-		ts.Fatalf("usage: setEnvValueFromFile envKey filePath")
-	}
-	srcRelPath = args[1]
-	envKey := args[0]
+func saveFileCmd(
```

Comment:
I can see the need for some supporting command written in go since testscript is so limited. I'm not sure about the baked in "saveDir" value. I sketched out another idea here https://github.com/blocky/attestation-service-examples/compare/feature/combo-test-bky-6885...temp/discuss-pr, maybe we should meet up and discuss?